namd/src/colvarproxy_namd.C

```diff
@@ -1539,6 +1551,143 @@ int colvarproxy_namd::smp_colvars_loop()
   return cvm::get_error();
 }
 
+struct cvc_info {
+  bool enabled;
+  bool parent_colvar_total_force_calc;
+  bool parent_colvar_Jacobian;
+  std::shared_ptr<colvar::cvc> cvc_ptr;
+};
+
+// TODO: Too bad! We have to traverse the AST twice!
+void cvc_max_depth(
+  const std::vector<std::shared_ptr<colvar::cvc>>& cvcs,
+  std::unordered_map<std::shared_ptr<colvar::cvc>, int>& cvc_depth_map,
+  int current_level = 0) {
+  for (auto it = cvcs.begin(); it != cvcs.end(); ++it) {
+    auto map_it = cvc_depth_map.find(*it);
+    if (map_it == cvc_depth_map.end()) {
+      cvc_depth_map.insert({*it, current_level});
+    } else {
+      if (map_it->second < current_level) {
+        map_it->second = current_level;
+      }
+    }
+    cvc_max_depth((*it)->children_cvcs(), cvc_depth_map, current_level+1);
+  }
+}
+
+void flatten_all_cvc_tree(
+  std::shared_ptr<colvar::cvc>& parent,
+  std::unordered_map<std::shared_ptr<colvar::cvc>, cvc_info>& cvc_info_map,
+  bool parent_colvar_total_force_calc,
+  bool parent_colvar_Jacobian) {
+  auto children = parent->children_cvcs();
+  for (auto it = children.begin(); it != children.end(); ++it) {
+    flatten_all_cvc_tree(*it, cvc_info_map, parent_colvar_total_force_calc, parent_colvar_Jacobian);
+  }
+  auto it_find = cvc_info_map.find(parent);
+  if (it_find == cvc_info_map.end()) {
+    cvc_info_map.insert({parent, cvc_info{
+      // TODO: Here calc_cvc_values calls cvcs[i]->is_enabled() , which is is_enabled(int f = f_cv_active)
+      //       I know both f_cv_active and f_cvc_active are 0 but are they the same option??
```

Comment:
The fact that `f_cv_active` and `f_cvc_active` have the same numerical value has no consequence, they should never be used in the same context, because they are respectively only meaningful in a `colvar` or `cvc` object. The `colvardeps` data of these two classes are non-overlapping. The relationship between them is a vertical (parent-child) dependency.

If we were to merge the two levels, then these two features would merge.

Reply:
The issue is that in https://github.com/Colvars/colvars/blob/0f2d6829a8ceb6e0af9335e46ffa5cb16bd7120c/src/colvar.cpp#L1437, `is_enabled()` is called, and since there is no function parameter passed, so I think it would call https://github.com/Colvars/colvars/blob/0f2d6829a8ceb6e0af9335e46ffa5cb16bd7120c/src/colvardeps.h#L167
which checks `f_cv_active` instead of `f_cvc_active`. My code is to follow what the original calls in `calc_cvc_values` and I am not sure if I need to follow it the same way.

Reply:
You're right, sorry. That is somewhat sloppy writing that I didn't remember well. It does rely on the "active" property being number 0.
https://github.com/Colvars/colvars/blob/0f2d6829a8ceb6e0af9335e46ffa5cb16bd7120c/src/colvardeps.h#L224

But a class inheriting from `colvardeps` could also override `is_enabled()` and change this convention.

Reply:
How much effort do you think it would be to remove the default argument for this virtual function? Remember that this is one of the "issues" that `clang-tidy` was complaining about.

Reply:
That would be very little effort. I'm happy to do that if that helps in any way.